DataBase Schema edited to match the new requirments

Ahmed-Eldera opened the pull request.

Ahmed-Eldera committed:
db new schema

Ahmed-Eldera committed:
fixed flyway and the project builds

Zeyad-F16 committed:
fix the tests

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/controller/AssetTypeController.java:
```diff
@@ -20,8 +20,8 @@ public AssetTypeController(AssetTypeService typeService) {
     }
 
     @GetMapping("/types")
```
can you currently move this api and the same for category in the asset controller w don't need to have separate controllers for them as they are child to assets and part of it so for types the api will be /assets/types
for category /assets/categories

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/controller/CategoryController.java:
```diff
@@ -20,8 +20,8 @@ public CategoryController(CategoryService categoryService) {
     @GetMapping
-    @PreAuthorize("hasAuthority('Admin')")
-    public List<Category> getAllCategories() {
+    @PreAuthorize("hasAuthority('ADMIN')")
```
same delete category controller and move this api to asset controller

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/dto/AssetDto.java:
```diff
@@ -14,7 +13,12 @@ public class AssetDto {
+    private String serialNumber;
+    private String purchaseDate;
+    private String warrantyEndDate;
+    private String status;
```
please make status enum not string and this is applies  for any status

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/entity/Asset.java:
```diff
@@ -1,45 +1,59 @@
-
-    @Id
-    @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private Long assetId;
```
rename into id for all entitiies

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/entity/AssetCategory.java:
```diff
@@ -1,18 +1,18 @@
+public class AssetCategory {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long categoryId;
```
rename into only id

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/entity/AssetCategory.java:
```diff
@@ -1,18 +1,18 @@
+    private Long categoryId;
+
+    @Column(name = "category_name", nullable = false, unique = true)
+    private String categoryName;
```
rename into name

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/entity/AssetStatus.java:
```diff
@@ -1,7 +1,8 @@
 
 public enum AssetStatus {
-    ACTIVE,
+    Available,
```
rename into AVAILABLE and ASSIGNED

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/entity/RequestStatus.java:
```diff
@@ -0,0 +1,7 @@
+package org.orange.oie.internship2025.assetmanagementsystem.entity;
+
+public enum RequestStatus {
+    Pending,
```
use upper case for enum values PENDING ACCEPTED REJECTED

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/entity/Role.java:
```diff
@@ -1,19 +1,19 @@
+public class Role {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long roleId;
```
rename into id

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/entity/Role.java:
```diff
@@ -1,19 +1,19 @@
+    private Long roleId;
+
+    @Column(name = "role_name", nullable = false, unique = true)
+    private String roleName;
```
rename into name

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/service/AssetTypeServiceImpl.java:
```diff
@@ -16,7 +16,7 @@ public AssetTypeServiceImpl(TypeRepository typeRepository) {
     }
 
     @Override
-    public List<Type> getAllTypes() {
```
please add 2 packages under the service package one for interface and other impl and remove files accordingly

Nourhan-Hesham commented on src/main/resources/application.properties:
```diff
@@ -1,24 +1,19 @@
+# ----------------------------------------
+# Database connection
+# ----------------------------------------
+spring.datasource.url=jdbc:mariadb://0.0.0.0:3302/ams
```
what is the ^M at the end of each line

Nourhan-Hesham commented on src/main/resources/db/migration/V2__mock_data.sql:
```diff
@@ -0,0 +1,70 @@
+                                               (2, 'LG UltraWide');
+
+-- Printers
+INSERT INTO asset_type (category_id, name) VALUES
```
please remove this file

Zeyad-F16 committed:
edits

Ahmed-Eldera committed:
dbInit bean intial implementation

Ahmeedsameer committed:
edits

Ahmeedsameer committed:
fixin department null in user

Ahmeedsameer committed:
BE commits for login and user lists according to new Db

Ahmed-Eldera committed:
ListAsset inital

Ahmeedsameer committed:
testing user list edits

Zeyad-F16 committed:
auth test

Commit:
delete old tests

Commit:
add test container and dbunit

Commit:
add transactional annotation on service

Commit:
add integration tests

Commit:
fix tests

Commit:
fix typo

Nourhan-Hesham committed:
Merge pull request #15 from EquantEgypt/fixing_usersList_and_login

Fixing users list and login

Commit:
remove duplicate dependency

Zeyad-F16 committed:
request

Commit:
fix jacoco unused packages

Commit:
fix tests

Commit:
Merge branch 'new_db_schema' of https://github.com/EquantEgypt/asset-management-system into new_db_schema

Commit:
refactor asset request to make asset optional

Ahmed-Eldera committed:
intial api working

# Conflicts:
#	src/main/java/org/orange/oie/internship2025/assetmanagementsystem/entity/AssetRequest.java
#	src/main/resources/db/migration/V1__create_ams_tables.sql

# Conflicts:
#	.gitignore

Zeyad-F16 committed:
edit asset type and add validation

Ahmed-Eldera committed:
added role conditions in service & testing

Maryiamreda committed:
Asset Assigment

Ahmed-Eldera committed:
fixed filtering by user

Maryiamreda committed:
fix integration test setup and authentication issues

Zeyad-F16 committed:
demo

Zeyad-F16 committed:
demo

Maryiamreda committed:
refactor:move assignment creation logic out of service function

Maryiamreda committed:
refactor asset assignment: fix mapper and improve response handling

Ahmed-Eldera committed:
packaged request and response dto to avoid conflict and enhace readability

Ahmed-Eldera committed:
added note

Maryiamreda committed:
add endpoints to fetch types by category and available assets by type

Zeyad-F16 committed:
edit add asset for demo

Ahmed-Eldera committed:
all done

Ahmed-Eldera committed:
all done

Zeyad-F16 committed:
add serial number for mini asset

Ahmed-Eldera committed:
added IT in role filtering for listAssets

Zeyad-F16 committed:
for tests

Zeyad-F16 committed:
remove mock data file

Ahmed-Eldera committed:
edits done

Ahmed-Eldera committed:
edits done

Ahmed-Eldera committed:
validations cleaned:removed unnecassary stuff

Maryiamreda committed:
ensure asset matches provided type and category before assignment

Nourhan-Hesham committed:
Merge pull request #17 from EquantEgypt/feature/assign-asset

Feature/assign asset

Ahmed-Eldera committed:
Fix:username not returning in request dto

Ahmed-Eldera committed:
fixed pipeline

Ahmed-Eldera committed:
rebased successfully

Nourhan-Hesham committed:
Merge pull request #19 from EquantEgypt/feature/createRequest

Feature/create request

Zeyad-F16 committed:
update asset

Ahmed-Eldera committed:
Fix:Return full AssetCategory in ListAssetDTO

Zeyad-F16 committed:
work nearly done , rebasing

Zeyad-F16 committed:
update asset after rebase

Zeyad-F16 committed:
delete mock data

Zeyad-F16 committed:
edit tests

Maryiamreda committed:
add category response fields and implement status/type filtering with specifications

Zeyad-F16 committed:
add a history for updated asset

Maryiamreda committed:
add  search functionality to requests

Maryiamreda committed:
Demo update

Zeyad-F16 committed:
apply comments

Ahmed-Eldera committed:
working with front end

# Conflicts:
#	src/main/java/org/orange/oie/internship2025/assetmanagementsystem/controller/AssetController.java

Ahmeedsameer committed:
User Details After change

Ahmeedsameer committed:
After make Admin can see user details

Zeyad-F16 committed:
apply comments-2

Zeyad-F16 committed:
edit mapper

Nourhan-Hesham committed:
Merge pull request #21 from EquantEgypt/feature/update-asset

Feature/update asset

Maryiamreda committed:
tests

Ahmeedsameer committed:
User Details After editing

Ahmed-Eldera committed:
case employee added in asset spec

Maryiamreda committed:
change controller design

Nourhan-Hesham committed:
Merge pull request #20 from EquantEgypt/feature/createRequest

fix needed for front end , returning full category

Nourhan-Hesham committed:
Merge pull request #25 from EquantEgypt/feature/users-details

Feature/users details

Maryiamreda committed:
conflicts

Malak-Ziad committed:
new branch for asset details with tests and history (#30)

Maryiamreda committed:
review comments done

Maryiamreda committed:
relocate assign asset tests to AssetControllerTest

Maryiamreda committed:
remove mock data

Nourhan-Hesham committed:
Merge pull request #31 from EquantEgypt/feature/handle-requests

Malak-Ziad committed:
final new branch for asset details and history

Zeyad-F16 committed:
Merge branch 'new_db_schema' into feature/asset_details-history

Zeyad-F16 committed:
resolve conflicts

Nourhan-Hesham committed:
Merge pull request #34 from EquantEgypt/feature/asset_details-history
